src/configuration.ts

```diff
@@ -8,15 +8,35 @@ export class VendorPortalApi {
 
     async client(): Promise<httpClient.HttpClient> {
       const http = new httpClient.HttpClient()
-      const replicatedEndpoint= this.endpoint;
+
+      const headers = {
+        "Authorization": this.apiToken,
+        "Content-Type": "application/json",
+        "Accept": "application/json"
+      };
+
+      // while this is specifically a github action, we still check
+      // for the github actions environment variables
+      if (process.env.CI) {
+        headers["X-Replicated-CI"] = "true";
+      }
+      if (process.env.GITHUB_RUN_ID) {
+        headers["X-Replicated-GitHubRunID"] = process.env.GITHUB_RUN_ID;
+      }
+      if (process.env.GITHUB_RUN_NUMBER) {
+        headers["X-Replicated-GitHubRunNumber"] = process.env.GITHUB_RUN_NUMBER;
+      }
+      if (process.env.GITHUB_SERVER_URL) {
+        headers["X-Replicated-GitHubServerURL"] = process.env.GITHUB_SERVER_URL;
+      }
+      if (process.env.GITHUB_REPOSITORY) {
+        headers["X-Replicated-GitHubRepository"] = process.env.GITHUB_REPOSITORY;
+      }
```

Comment:
Thinking these should be set in the [replicated-actions](https://github.com/replicatedhq/replicated-actions/blob/main/create-cluster/src/index.ts#L23-L28). So that if we reuse this sdk for other CI's or IDE integration...
But for now, we only use it as part of GH actions (afaik), so 👍

Reply:
Why would these be set in actions? This is the library that creates the http client, i'd like to have them always set, regardless of who's invoking this functionality. If we use this in a NodeJS SDK, it should also set these variables.